tests/common2/bgp_route_control.py

```diff
@@ -0,0 +1,255 @@
+        logger.debug("BGP bulk command: %s", command)
+
+        try:
+            response = requests.post(url, data=data, timeout=90)
+            if response.status_code != 200:
+                raise AssertionError(
+                    f"HTTP request failed with status {response.status_code}. URL: {url}. Data: {data}"
+                )
+        except requests.RequestException as e:
+            raise AssertionError(f"HTTP request to ExaBGP API failed: {e}") from e
+
+    @staticmethod
+    def update_route_with_attributes(action: str, ptfip: str, port: int, route_dict: Dict[str, Any]) -> None:
+        """
+        Update a route with custom attributes using the bgp_helpers format.
+
+        Args:
+            action: "announce" or "withdraw"
+            ptfip: PTF host IP address
+            port: ExaBGP HTTP API port
+            route_dict: Dictionary with route attributes like:
+                       {"prefix": "10.1.1.0/24", "nexthop": "10.1.1.1", "community": "1010:1010"}
+        """
+        if action not in ["announce", "withdraw"]:
+            raise ValueError(f"Unsupported route update operation: {action}")
+
+        if "prefix" not in route_dict or "nexthop" not in route_dict:
+            raise ValueError("route_dict must contain 'prefix' and 'nexthop' keys")
+
+        # Build message in bgp_helpers format
+        msg = f'{action} route {route_dict["prefix"]} next-hop {route_dict["nexthop"]}'
+
+        if "community" in route_dict:
+            msg += f' community {route_dict["community"]}'
+
+        if "local_preference" in route_dict:
+            msg += f' local-preference {route_dict["local_preference"]}'
+
+        url = f"http://{ptfip}:{port}"
+        data = {"commands": msg}
+
+        logger.info("BGP update route: URL=%s, Data=%s", url, data)
+
+        try:
+            response = requests.post(url, data=data, timeout=30)
+            if response.status_code != 200:
+                raise AssertionError(f"HTTP request failed with status {response.status_code}")
+        except requests.RequestException as e:
+            raise AssertionError(f"HTTP request to ExaBGP API failed: {e}") from e
+
+
+# Convenience functions for backward compatibility
+def announce_route(
+    ptfip: str, neighbor: str, route: str, nexthop: str, port: int, community: Optional[str] = None
+) -> None:
+    """
+    Convenience function for announcing a single route (test_bgp_speaker format).
+    """
+    BGPRouteController.announce_route(ptfip, neighbor, route, nexthop, port, community)
+
+
+def withdraw_route(
+    ptfip: str, neighbor: str, route: str, nexthop: str, port: int, community: Optional[str] = None
+) -> None:
+    """
+    Convenience function for withdrawing a single route (test_bgp_speaker format).
+    """
+    BGPRouteController.withdraw_route(ptfip, neighbor, route, nexthop, port, community)
+
+
+def announce_route_with_community(
```

Comment:
Is there a special purpose to keep both announce_route_with_community and announce_route? Could we consider combining them into a single function? thanks

Reply:
The current design keeps both functions to maintain backward compatibility for different test requirements. `announce_route` is for basic BGP tests without local preference, while `announce_route_with_community` is specifically for BGP sentinel tests that require both community values and a local preference of 10000. I think combining them would reduce API clarity.